src/main/java/marquez/api/mappers/DatasourceResponseMapper.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package marquez.api.mappers;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.Collections;
+import java.util.List;
+import lombok.NonNull;
+import marquez.api.models.DatasourceResponse;
+import marquez.service.models.Datasource;
+
+public final class DatasourceResponseMapper {
+  private DatasourceResponseMapper() {}
+
+  public static DatasourceResponse map(@NonNull Datasource datasource) {
+    return new DatasourceResponse(
+        datasource.getCreatedAt().toString(),
+        datasource.getDataSourceName().getValue(),
+        datasource.getConnectionUrl().getRawValue());
+  }
+
+  public static List<DatasourceResponse> map(@NonNull List<Datasource> datasources) {
```

Comment:
You'll want to follow the recommended approach to return a list of responses: https://github.com/MarquezProject/marquez/issues/309

Reply:
Nice - much cleaner that way. Will be in next revision.

Reply:
Please see PRs #394, #387, and #384, where this is fixed.